### ISSUE TITLE
Model fails to distinguish "victim" and "perpetrator" in violence detection

### ISSUE THREAD

**Issue by Repet1** [iSEE-Laboratory/LLMDet#62]
I'm using this project to test image content safety detection, specifically for identifying violent elements in images.  

### Problem Description:  
When I run the model to detect "victim", "perpetrator", and "weapon" in violent scenes, the model cannot distinguish between "victim" and "perpetrator" — both labels are often misassigned or overlapping. This seems inconsistent with the expected capabilities of large models, so I suspect there might be an issue with my setup.  

### Command Used:  
```bash
python image_demo.py images/abuse2.jpeg configs/grounding_dino_swin_l.py --weight large.pth --text 'victim . Perpetrator . weapon .' -c --pred-score-thr 0.3
```  

### Expected Outcome:  
The model should correctly label the victim (the person being harmed) and perpetrator (the person committing violence) in the image.  

### Actual Outcome:  
Labels for "victim" and "perpetrator" are often swapped or applied to the same object, with no clear distinction.  

Could this be due to incorrect text prompts, threshold settings, or model configuration? Any guidance on adjusting the parameters would be greatly appreciated!  

Thank you for your help.

**Comment by fushh** [comment]
I have read the email and I think the results are reasonable and you do not make any mistakes.
Possible reason is that "victim", "perpetrator" are not in our pretraining dataset.
I think you can use a Large Vision Language Model (like Qwen2.5-VL) to do this task or use a custom dataset to fine-tune our model.
